students/johnpharmd/lesson03/strformat_lab.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+
+# started with form_st = 'file_{:d}: {:f}, {:d}, {:f}'
+# Internet search pointed me to https://docs.python.org/2/library/string.html,
+# which, apparently, is not (completely) deprecated in Python 3.
+# Also used https://docs.python.org/3.3/library/string.html#formatstrings.
+# iteration: form_st = 'file_{:03d}: {:f}, {:e}, {:e}'
+# iteration: form_st = 'file_{:03d}: {:.2f}, {:e}, {:e}'
+# iteration: form_st = 'file_{:03d}: {:.2f}, {:.2e}, {:e}'
+# Task 1
+ form_st = 'file_{:03d}: {:.2f}, {:.2e}, {:.2e}'
+
+ # Task 2
+ # started with:
+ # form_st2 = 'file_{integer}: {flo}, {expo1}, {expo2}'
+ # Found this:
+ # https://stackoverflow.com/questions/3228865/how-do-i-format-a-number-with-a-variable-number-of-digits-in-python
+ form_st1 = 'file_{num:0{width}}: {flo:.2f}, {expo1:.2e}, {expo2:.2e}'
+
```

Comment:
Call format method on this string as well. Not sure if you've gotten to this in the course materials yet, but you can pass a dict to the format method in the same way that you do below with a tuple, so this can be written like this:

```
numbers = {
    'num': 2,
    'flo': 123.4567,
    'expo1': 10000,
    'expo2': 12345.67,
}

form_st1 = 'file_{num:03d}: {flo:.2f}, {expo1:.2e}, {expo2:.2e}'.format(**numbers)
```